src/decorators/docs/course.decorator.ts

```diff
@@ -1,281 +1,63 @@
-    }),
-    ApiQuery({
-      name: 'cursorId',
+      name: 'keyword',
       required: false,
-      type: 'number',
-    }),
-    ApiResponse({
-      status: 200,
-      description: '교양 과목명으로 강의 검색 성공 시',
-      type: PaginatedCoursesDto,
-    }),
-  ],
-  searchMajorProfessorName: [
-    ApiBearerAuth('accessToken'),
-    ApiOperation({
-      summary: '전공 과목 담당 교수님 성함으로 강의 검색',
-      description: '전공 과목 담당 교수님 성함을 입력하여 강의를 검색합니다.',
-    }),
-    ApiQuery({
-      name: 'major',
-      required: true,
-      type: 'string',
-    }),
-    ApiQuery({
-      name: 'professorName',
-      required: true,
       type: 'string',
     }),
     ApiQuery({
-      name: 'cursorId',
+      name: 'classification',
       required: false,
-      type: 'number',
-    }),
-    ApiResponse({
-      status: 200,
-      description: '전공 과목 담당 교수님 성함으로 강의 검색 성공 시',
-      type: PaginatedCoursesDto,
-    }),
-    ApiKukeyExceptionResponse(['MAJOR_REQUIRED']),
-  ],
-  searchGeneralProfessorName: [
-    ApiBearerAuth('accessToken'),
-    ApiOperation({
-      summary: '교양 담당 교수님 성함으로 강의 검색',
-      description: '교양 담당 교수님 성함을 입력하여 강의를 검색합니다.',
-    }),
-    ApiQuery({
-      name: 'professorName',
-      required: true,
       type: 'string',
     }),
-    ApiQuery({
-      name: 'cursorId',
-      required: false,
-      type: 'number',
-    }),
-    ApiResponse({
-      status: 200,
-      description: '교양 담당 교수님 성함으로 강의 검색 성공 시',
-      type: PaginatedCoursesDto,
-    }),
-  ],
-  getGeneralCourses: [
-    ApiBearerAuth('accessToken'),
-    ApiOperation({
-      summary: '교양 강의 조회',
-      description: '모든 교양 강의를 조회합니다.',
-    }),
-    ApiQuery({
-      name: 'cursorId',
-      required: false,
-      type: 'number',
-    }),
     ApiResponse({
       status: 200,
-      description: '교양 강의 조회 성공 시',
-      type: PaginatedCoursesDto,
-    }),
-  ],
-  getMajorCourses: [
-    ApiBearerAuth('accessToken'),
-    ApiOperation({
-      summary: '전공 강의 조회',
-      description: '해당 과의 모든 전공 강의를 조회합니다.',
-    }),
-    ApiQuery({
-      name: 'major',
-      required: true,
-      type: 'string',
-    }),
-    ApiQuery({
-      name: 'cursorId',
-      required: false,
-      type: 'number',
-    }),
-    ApiResponse({
-      status: 200,
-      description: '전공 강의 조회 성공 시',
+      description: '강의 검색 성공 시',
       type: PaginatedCoursesDto,
     }),
     ApiKukeyExceptionResponse(['MAJOR_REQUIRED']),
-  ],
-  getAcademicFoundationCourses: [
-    ApiBearerAuth('accessToken'),
-    ApiOperation({
-      summary: '학문의 기초 강의 조회',
-      description: '해당 단과대의 모든 학문의 기초 강의를 조회합니다.',
-    }),
-    ApiQuery({
-      name: 'college',
-      required: true,
-      type: 'string',
-    }),
-    ApiQuery({
-      name: 'cursorId',
-      required: false,
-      type: 'number',
-    }),
-    ApiResponse({
-      status: 200,
-      description: '학문의 기초 강의 조회 성공 시',
-      type: PaginatedCoursesDto,
-    }),
     ApiKukeyExceptionResponse(['COLLEGE_REQUIRED']),
```

Comment:
```suggestion
    ApiKukeyExceptionResponse(['MAJOR_REQUIRED', 'COLLEGE_REQUIRED']),
```
요것도 사소한건데 한 배열에 넣어주면 swagger 상에 모든 에러 목록이 잘 표시될 것 같습니다!